Convert GOvis.Rnw to GOvis.Rmd

sonali8434 opened the pull request:
@Bioconductor/sweave2rmd please review my contribution for Outreachy.
Hello @jwokaty , This is my pull request for the second contribution task for Outreachy. There are some differences from the original PDF as the graphs are showing in the middle of code blocks, which is not technically possible. So, I followed the .Rnw file and made the necessary changes. I also faced a problem during the conversion process, which I discussed in Slack. The issue was with the "@GentGO2004a" citation, which was not available in the .bib file. As suggested by @jwokaty in Slack, I changed the citation accordingly. 
I have reviewed the rest of the changes myself.

Original pdf-[PDF](https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOvis.pdf)
Here is the HTML link- [HTML Link](https://rpubs.com/Sonali8434/1019846)  for your reference Thank you.

sonali8434 committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

sonali8434 committed:
Convert file to Rmd

sonali8434 committed:
Remove Govis.Rnw

sonali8434 committed:
Update GOvis.Rmd

Made changes after reviewing the file myself by comparing it with original pdf.

sonali8434 committed:
Update GOvis.Rmd

Corrected the citation with the reference given by Jennifer and also changed some formatting

sonali8434 committed:
Update GOvis.Rmd

Added {#introduction .unnumbered} as there is no number in original pdf.

sonali8434 committed:
Update GOvis.Rmd

Added eval=FALSE in code chunk since the image is not shown in pdf

sonali8434 committed:
Update GOvis.Rmd

Corrected few formatting and also adjusted the picture place according to pdf in the best possible way.

sonali8434 committed:
Update GOvis.Rmd

added link-citations: yes so that we can link to references

sonali8434 committed:
Update GOvis.Rmd

Changes citation brackets after comparing from the pdf.

sonali8434 committed:
Update GOvis.Rmd

Corrected the formatting and made the final changes.

sonali8434 committed:
Update GOvis.Rmd

Added the missing %\VignetteEngine{knitr::rmarkdown} line in the YAML header .

sonali8434 committed:
Update GOvis.Rmd

changed VignetteIndexEntry to make it same as the title

jwokaty commented on vignettes/GOvis.Rmd:
```diff
@@ -0,0 +1,547 @@
+make some of the
+concepts concrete and to provide extensive, but related
+examples we will make
+use of a microarray data set [@Chiaretti2004].
```
Can I ask you to reformat sections that are like these so that they are easier to read?

sonali8434 replied:
Ohk I will do that for all the paragraphs.

sonali8434 committed:
Update GOvis.Rmd

Reformat the paragraph so that they will be easier to read.

sonali8434 committed:
Update GOvis.Rmd

Reformat code blocks( as per pdf representation).

sonali8434 commented:
@jwokaty I have done the formatting asked by you and done some more formatting changes in code chunks like line spacing according to pdf. I think now my PR is ready for review since all checklist are now done from my side, Please let me know if any changes are required I have seen that you advised one of the contributor about removing {#introduction .unnumbered} but I have included it since it is necessary to match the content of pdf.

If it should be removed please let me know I will do that, Thank you.

sonali8434 committed:
Update GOvis.Rmd

remove extra CR line that was missed.

sonali8434 committed:
Update GOvis.Rmd

Corrected the label of figures.

sonali8434 commented:
@jwokaty I have made the requested formatting changes and also made additional modifications to the code chunks, such as adjusting line spacing to match the format of the PDF. I believe that my pull request is now ready for your review as I have completed all the items on the checklist. However, if you have any suggestions or require any further changes, please let me know. I noticed that you advised another contributor to remove the {#introduction .unnumbered} tag, but I have included it as it is necessary to match the content of the PDF. If you feel that it should be removed, please inform me and I will make the necessary changes. Thank you for your time and consideration.

sonali8434 committed:
Update GOvis.Rmd

Removed extra space from caption.

sonali8434 commented:
Here is the checklist for this review 

 the .Rmd file knits to HTML
 R CMD build runs without errors or timeouts
 the tarball from R CMD build contains the HTML (check with the
following by substituting the package name and vignette name tar ztf package_name.tar.gz | grep 'doc/vignette_name')
 the .Rnw file has been removed
 in the DESCRIPTION, BiocStyle and knitr are listed in Suggests
 in the DESCRIPITON, the line VignetteBuilder: knitr exists
 any added lines use the same spacing and indents as the existing
document
 if agreed with the Maintainer, the contributor is in the author list
in the DESCRIPTION file; refer to the contribution guide for acceptable
formats
 if agreed with the Maintainer, the contributor is in the author list
in the vignette's YAML; refer to the contribution guide for acceptable formats
 if this pull request includes a conversion from using the separate Author
and Maintainer lines into the Authors@R vector, check that the
Authors@R vector includes the maintainer as specified with role='cre'.
 if this pull request includes a conversion from using the separate Author
and Maintainer lines into the Authors@R vector, the Maintainer line
is completely removed.
 HTML document is representative of the PDF in content and
in general the presentation
 Where the contributor was not able to preserve the content and presentation
of the PDF is noted as a comment in the pull request
 the R Markdown file is representative of the Sweave document and follows
best practices, such as replacing links to Bioconductor packages with calls
to BiocStyle's Biocpkg()

sonali8434 commented:
Please find below a checklist for the pull request, which you can use to review my work. I have reviewed it myself, and I welcome any feedback or suggestions you may have.

- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [x] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [x] any added lines use the same spacing and indents as the existing
  document
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

sonali8434 committed:
Formatted the paragraphs.

jwokaty commented on vignettes/GOvis.Rmd:
````diff
@@ -0,0 +1,497 @@
+```
+
+The results reported in this document are based on
+`r packageDescription("hgu95av2.db")[["Version"]]` version of the
+`r Biocpkg("hgu95av2.db")` data package and on
+`r packageDescription("GO.db")[["Version"]]` of the `r Biocpkg("GO.db")` package.
````
Use BiocStyle's `pkg_ver()` to get the version of packages. Can I also ask you to format this paragraph nicely?

sonali8434 replied:
Ohk I will do that.

jwokaty replied:
I'm in the middle of reviewing. I will send a message when the review is complete.

jwokaty commented on vignettes/GOvis.Rmd:
```diff
@@ -0,0 +1,497 @@
+conditions first they were included on the chip being used and second
+they have a GO annotation for the ontology of interest.
+
+<a id="sec:example" ></a>
```
I think this can be removed as well as any extra lines.

sonali8434 replied:
we have one paragraph in visualization section in first paragraph where we use this label "sec:example" if we remove this line then how can we get back to this section after we click on "0.3". ? I have added this line for this purpose only please clarify.. should I remove this or leave this?

sonali8434 replied:
Done.

jwokaty reviewed (changes requested):
This was well done. Good work on the conversion and also reviewing.

I left one note within the .Rmd. Additionally, I think we need to change the citation style because the HTML is a little different than the original, which I believe is published in a scientific journal; however, I have not located the journal yet so I can't tell you which one it is. We could just make a custom file to match it. You can see how to do this at https://bookdown.org/yihui/rmarkdown-cookbook/bibliography.html#changing-citation-style.

Lastly I noticed that the original document itself has inconsistent formatting for code. Sometimes I see 2 spaces for indents, other times 4. I also saw some of the code doesn't use general practices to help readers easily know if a line is part of a function's arguments. I leave this up to you if you'd like to help sort of any this out.

sonali8434 commented:
> This was well done. Good work on the conversion and also reviewing.
> 
> I left one note within the .Rmd. Additionally, I think we need to change the citation style because the HTML is a little different than the original, which I believe is published in a scientific journal; however, I have not located the journal yet so I can't tell you which one it is. We could just make a custom file to match it. You can see how to do this at https://bookdown.org/yihui/rmarkdown-cookbook/bibliography.html#changing-citation-style.
> 
> Lastly I noticed that the original document itself has inconsistent formatting for code. Sometimes I see 2 spaces for indents, other times 4. I also saw some of the code doesn't use general practices to help readers easily know if a line is part of a function's arguments. I leave this up to you if you'd like to help sort of any this out.

Ohk @jwokaty I will do some research and if I found any sort of solution which can be used to correct this I will discuss with you in slack and together we can find a solution for this.

jwokaty commented:
Regarding the bibliography, I asked around and the suggestion I got was to use the style associated with *Bioinformatics*. I think it's still a little different, but if you're not able to find something suitable, I think it will be fine.

sonali8434 commented:
> Regarding the bibliography, I asked around and the suggestion I got was to use the style associated with _Bioinformatics_. I think it's still a little different, but if you're not able to find something suitable, I think it will be fine.

Ohk @jwokaty  Thank you so much...

sonali8434 committed:
Added bioinformatics.csl

sonali8434 committed:
used BiocStyle's pkg_ver() and also wrap lines

sonali8434 commented:
Hello, @jwokaty I have made the requested changes .I have also reformat the code chunks which you can see in my commit... I will again go through the links given you in Slack and made further change if needed for now you can see my commit and guide me that if I am in the right direction regarding updating code chunks Thank you..

sonali8434 committed:
Fixed indents of code chunks.

sonali8434 commented:
@jwokaty I have updated the code chunks and made the final changes you can see my commit and let me know if any further changes are required.. 
Thank you so much and also I have left this https://github.com/Bioconductor/GOstats/pull/3#discussion_r1152433309 part as I am not sure about this Please review my work whenever you are free.

sonali8434 committed:
Removed <a id="sec:example" ></a> 

Removed <a id="sec:example" ></a>  line and also extra spaces.

sonali8434 commented:
@jwokaty, I have updated the code chunks and made final changes. You can see my commit and let me know if any further changes are required.

 The changes I have made are as follows:
1. used BiocStyle's pkg_ver( ) to get the version and wrapped the lines nicely.
2. I have changed the citation style to bioinformatics.csl.
3. I have formatted the code chunks according to coding style practices that help others understand, evaluate, and debug code more easily.
    - Specifically, I have used <- instead of = for assigning variables (except in function arguments).
    -  used 4 spaces instead of 2 spaces or tabs throughout the document.
    -  added spaces between "=", aligned function arguments, and made other necessary changes.

Thank you so much,  please review my work whenever you are free.

sonali8434 committed:
Fixed the inline r code

jwokaty commented:
Thanks for your contribution!

sonali8434 commented:
> Thanks for your contribution!

Thank you so much I have learned a lot during this contribution period
